runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentsClientBridge.java

```diff
@@ -244,9 +247,35 @@ public void onProgressChanged(int progress) {
     }
 
     @Override
-    public WebResourceResponse shouldInterceptRequest(String url) {
+    public XWalkWebResourceResponseInternal shouldInterceptRequest(
+            WebResourceRequestInner request) {
         if (isOwnerActivityRunning()) {
-            return mXWalkResourceClient.shouldInterceptLoadRequest(mXWalkView, url);
+            XWalkWebResourceResponseInternal xwalkResponse;
+            //For compatibility with the old shouldInterceptLoadRequest.
+            WebResourceResponse response =
+                 mXWalkResourceClient.shouldInterceptLoadRequest(mXWalkView, request.url);
+            if (response == null) {
+                xwalkResponse = mXWalkResourceClient.shouldInterceptLoadRequest(mXWalkView,
+                                        new XWalkWebResourceRequestHandlerInternal(request));
+                if (xwalkResponse == null) return null;
+
+                // XWalkWebResourceResponse should support null headers.
+                Map<String, String> responseHeaders = xwalkResponse.getResponseHeaders();
+                if (responseHeaders == null) responseHeaders = new HashMap<String, String>();
+
+                return new XWalkWebResourceResponseInternal(
+                        xwalkResponse.getMimeType(),
```

Comment:
return directly does't work, actually I don't know the reason yet, I also feel confuse. 
current solution can pass all the test, so let's investigate in the future.